lib/Core/Searcher.cpp

```diff
@@ -258,29 +258,37 @@ bool WeightedRandomSearcher::empty() {
   return states->empty(); 
 }
 
-///
-RandomPathSearcher::RandomPathSearcher(Executor &_executor)
-  : executor(_executor) {
-}
+RandomPathSearcher::RandomPathSearcher(PTree &_processTree)
+    : processTree(_processTree), idBitMask(_processTree.getNextId()) {}
 
 RandomPathSearcher::~RandomPathSearcher() {
 }
+// Check if n is a valid pointer and a node belonging to us
+#define IS_OUR_NODE_VALID(n)                                                   \
```

Comment:
"OUR"?

Reply:
The one we arw considering right now. That's just how I thought about it

Reply:
`CHILD_NODE`, `IS_CHILD_NODE`, `ACCESSIBLE_CHILD_NODE`, ...

Reply:
But it obviously is child node. The point is it is a node  that belongs to our subset of states and it is valid.

Reply:
It might be a child node in PTree but not a child node for RandomPath. Maybe sth. in the direction of `IS_MAINTAINED_CHILD`...

Reply:
I don't think MAINTAINED is a good word here. In my mind the searcher doesn't maintain anything. A node belongs to the searcher, ie. is part of the searcher or not.

Reply:
Besides that a node doesn't belong to a searcher... It is just ASSIGNED to or ASSOCIATED with it.

Reply:
Do we really need a macro here? A simple static function would do the same.